ur_calibration/README.md

```diff
@@ -0,0 +1,77 @@
+# ur_calibration
+
+Package for extracting the factory calibration from a UR robot and changing it to be used by `ur_description` to gain a correct URDF model.
+
+Each UR robot is calibrated inside the factory giving exact forward and inverse kinematics. To also
+make use of this in ROS, you first have to extract the calibration information from the robot.
+
+Though this step is not necessary, to control the robot using this driver, it is highly recommended
+to do so, as end effector positions might be off in the magnitude of centimeters.
+
+## Nodes
+### calibration_correction
+This node extracts calibration information directly from a robot, calculates the URDF correction and
+saves it into a .yaml file.
+
+In the launch folder of the ur_calibration package is a helper script:
+
+```bash
+$ ros2 launch ur_calibration calibration_correction.py.launch \
+robot_ip:=<robot_ip> target_filename:="${HOME}/my_robot_calibration.yaml"
+```
+
+For the parameter `robot_ip` insert the IP address on which the ROS pc can reach the robot. As
+`target_filename` provide an absolute path where the result will be saved to.
+
+## Creating a calibration / launch package for all local robots
+When dealing with multiple robots in one organization it might make sense to store calibration data
+into a package dedicated to that purpose only. To do so, create a new package (if it doesn't already
+exist)
+
+```bash
```

Comment:
This process is a bit too "hacky". We should provide an argument to use custom calibration package in launch files in this repo. Is this possible? Does it make sense?

Reply:
@destogl We could steer the user towards copying the calibration yaml file into appropriate subfolder of the `ur_description` package. This way we can make `kinematics_yaml` file an argument to the launch file

Reply:
If you ever release `Universal_Robots_ROS2_Driver`, so user can install using `apt` (or `yum` or whatever): where would `ur_description` be? And would it be world-writable?

Reply:
@gavanderhoorn This is very valid point. I think that the best solution is to have a bash/python-script generating the package with calibration data (long-term) and then referencing this through argument in the launch files.

Reply:
We've discussed quite a few of these topics when the ROS 1 driver was developed.

I'm not sure how exactly, but it might be good idea to reuse some of those design decisions / experiences here.

Reply:
Additional permission would be needed if the `ur_description` is installed via binaries. It is good to avoid that. Also, I think the script is not necessary and it hides from the user some steps that make quite clear what is happening...